test/staking/SwapHelper.test.ts

```diff
@@ -0,0 +1,403 @@
+  let pair: Contract, uniswapRouter: Contract, usdcPair, compPair;
+  let cDAI, cUsdc, usdc, cBat: Contract;
+  let gasFeeOracle,
+    daiEthOracle: Contract,
+    daiUsdOracle: Contract,
+    batUsdOracle: Contract,
+    ethUsdOracle: Contract,
+    compUsdOracle: Contract;
+  let goodReserve: Contract;
+  let goodCompoundStaking;
+  let goodFundManager: Contract;
+  let avatar,
+    controller,
+    founder,
+    staker,
+    schemeMock,
+    signers,
+    nameService,
+    setDAOAddress,
+    genericCall,
+    goodCompoundStakingFactory,
+    goodCompoundStakingTestFactory;
+
+  before(async () => {
+    [founder, staker, ...signers] = await ethers.getSigners();
+    schemeMock = signers.pop();
+    const cdaiFactory = await ethers.getContractFactory("cDAIMock");
+    const cBatFactory = await ethers.getContractFactory("cDecimalsMock");
+    const goodFundManagerFactory = await ethers.getContractFactory(
+      "GoodFundManager"
+    );
+    goodCompoundStakingFactory = await getStakingFactory("GoodCompoundStaking");
+    goodCompoundStakingTestFactory = await getStakingFactory(
+      "GoodCompoundStakingTest"
+    );
+
+    const uniswap = await deployUniswap();
+    uniswapRouter = uniswap.router;
+    const { factory, weth } = uniswap;
+
+    const daiFactory = await ethers.getContractFactory("DAIMock");
+    let {
+      controller: ctrl,
+      avatar: av,
+      gd,
+      identity,
+      nameService: ns,
+      setDAOAddress: sda,
+      setSchemes,
+      marketMaker: mm,
+      daiAddress,
+      cdaiAddress,
+      reserve,
+      setReserveToken,
+      genericCall: gc,
+      COMP,
+    } = await createDAO();
+
+    genericCall = gc;
+    dai = await ethers.getContractAt("DAIMock", daiAddress);
+    cDAI = await ethers.getContractAt("cDAIMock", cdaiAddress);
+    comp = COMP;
+    avatar = av;
+    controller = ctrl;
+    setDAOAddress = sda;
+    nameService = ns;
+    goodReserve = reserve;
+    console.log("deployed dao", {
+      founder: founder.address,
+      gd,
+      identity,
+      controller,
+      avatar,
+    });
+    goodFundManager = await upgrades.deployProxy(
+      goodFundManagerFactory,
+      [nameService.address],
+      {
+        kind: "uups",
+      }
+    );
+    await setDAOAddress("FUND_MANAGER", goodFundManager.address);
+    console.log("Deployed goodfund manager", {
+      manager: goodFundManager.address,
+    });
+
+    console.log("deployed contribution, deploying reserve...", {
+      founder: founder.address,
+    });
+
+    console.log("setting permissions...");
+    const tokenUsdOracleFactory = await ethers.getContractFactory(
+      "BatUSDMockOracle"
+    );
+    setDAOAddress("UNISWAP_ROUTER", uniswapRouter.address);
+
+    daiUsdOracle = await tokenUsdOracleFactory.deploy();
+    const compUsdOracleFactory = await ethers.getContractFactory(
+      "CompUSDMockOracle"
+    );
+    compUsdOracle = await compUsdOracleFactory.deploy();
+    goodCompoundStaking = await deployStaking(
```

Comment:
aren't you creating a new goodcompount staking in each test?

Reply:
yes removed it